src/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseException.cs

```diff
@@ -72,6 +73,8 @@ protected LicenseException(SerializationInfo info, StreamingContext context) : b
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             base.GetObjectData(info, context);
+            info.AddValue("type", null); // Type is not serializable.
+            info.AddValue("instance", _instance);
```

Comment:
_instance is Object. We made the decision to optimistically serialize object fields. But should we be try/catching around AddValue in case it's not serializable -- so we can serialize null instead and they still get a successful serialization with the other fields?

Reply:
Conversely are there any places we do GetValue on fields that are of Object type? Should those be tolerant to receiving a type that we can't deserialize?

Reply:
The issue with that (and tbh I wasn't aware of that behavior) is that if the serialization blob contains a field/type that isn't serializable the whole deserialization fails with a SerializationException. Because of that it wouldn't help surrounding it with a try catch or use the helper method.

Reply:
Ah, even if we never call AddValue for it? Well, let's hope there aren't such cases. That also means it makes sense for us to always call AddValue since if we get that far then it should work.

Reply:
The ones that don't work from Desktop --> Core are mentioned in my excel and in the initial post. If we would add the AddValue command with non-serializable types Core --> Core serialization would also be broken. With the current behavior only Desktop --> Core is broken for some types.